src/views/Editor/index.js

```diff
@@ -66,6 +74,35 @@ var EditorPage = React.createClass({
         this.forceUpdate()
     },
 
+    componentWillUnmount() {
+        window.removeEventListener("beforeunload", this.handler)
+        this.props.dispatch(setValidationErrors({}))
+    },
+
+    routerWillLeave(nextLocation) {
+        if (this.state.isDirty) {
+            return 'Changes that you have made may not be saved.\n\nAre you sure you want to leave?'
```

Comment:
This should be in Finnish, as the form is currently in Finnish only. No internationalization support for the strings needed at the moment.